source/snapshot/fetch_worker.go

```diff
@@ -257,11 +257,10 @@ func (f *FetchWorker) updateSnapshotEnd(ctx context.Context, tx pgx.Tx) error {
 		return nil
 	}
 
-	if err := tx.QueryRow(
-		ctx,
-		fmt.Sprintf("SELECT max(%s) FROM %s", f.conf.Key, f.conf.Table),
-	).Scan(&f.snapshotEnd); err != nil {
-		return fmt.Errorf("failed to query max on %q.%q: %w", f.conf.Table, f.conf.Key, err)
+	query := fmt.Sprintf("SELECT COALESCE(max(%s), 0) FROM %s", f.conf.Key, f.conf.Table)
+	err := tx.QueryRow(ctx, query).Scan(&f.snapshotEnd)
```

Comment:
nit, why not inline the err in the if, to not have to deal with scope

Reply:
Makes sense, will do.